server/data/data-jmap/src/main/java/org/apache/james/jmap/mime4j/SizeUtils.java

```diff
@@ -0,0 +1,99 @@
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.mime4j;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.apache.james.mime4j.dom.Body;
+import org.apache.james.mime4j.dom.Entity;
+import org.apache.james.mime4j.dom.SingleBody;
+import org.apache.james.mime4j.message.BodyPart;
+import org.apache.james.mime4j.message.DefaultMessageWriter;
+import org.apache.james.mime4j.message.MessageImpl;
+
+import com.google.common.io.CountingOutputStream;
+
+public class SizeUtils {
+    public static long sizeOf(Entity entity) throws IOException {
+        if (entity instanceof BodyPart) {
+            BodyPart bodyPart = (BodyPart) entity;
+
+            return sizeOf(bodyPart.getBody());
+        }
+        if (entity instanceof MessageImpl) {
+            MessageImpl bodyPart = (MessageImpl) entity;
+            CountingOutputStream countingOutputStream = new CountingOutputStream(OutputStream.nullOutputStream());
+            DefaultMessageWriter defaultMessageWriter = new DefaultMessageWriter();
+            defaultMessageWriter.writeHeader(bodyPart.getHeader(), countingOutputStream);
+
+            return sizeOf(bodyPart.getBody());
+        }
+        CountingOutputStream countingOutputStream = new CountingOutputStream(OutputStream.nullOutputStream());
+        DefaultMessageWriter defaultMessageWriter = new DefaultMessageWriter();
+        defaultMessageWriter.writeEntity(entity, countingOutputStream);
+        return countingOutputStream.getCount();
+    }
+
+    public static long sizeOf(Body body) throws IOException {
+        if (body instanceof FakeBinaryBody) {
+            return ((FakeBinaryBody) body).getSize();
+        }
+        if (body instanceof SingleBody) {
+            return ((SingleBody) body).size();
+        }
+        System.out.println(body.getClass());
+        CountingOutputStream countingOutputStream = new CountingOutputStream(OutputStream.nullOutputStream());
+        DefaultMessageWriter defaultMessageWriter = new DefaultMessageWriter();
+        defaultMessageWriter.writeBody(body, countingOutputStream);
+        return countingOutputStream.getCount();
+    }
+
+    /*
+      private def size(entity: Entity): Try[Size] = {
```

Comment:
Commented code block spotted

Reply:
Can be removed.

Sorry I was in a hurry